reana_client/cli/common.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of REANA.
+# Copyright (C) 2018 CERN.
+#
+# REANA is free software; you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation; either version 2 of the License, or (at your option) any later
+# version.
+#
+# REANA is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# REANA; if not, write to the Free Software Foundation, Inc., 59 Temple Place,
+# Suite 330, Boston, MA 02111-1307, USA.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization or
+# submit itself to any jurisdiction.
+"""Cli functionality shared between multiple modules of REANA-Client."""
+
+import click
+
+
+def workflow_uuid_or_name(ctx, param, value):
+    """Get UUID of workflow from configuration / cache file based on name."""
+    if not value:
+        click.echo(click.style(
+            'Workflow name must be provided either with '
+            '`--workflow` option or with `$REANA_WORKON` '
+            'environment variable', fg='red'),
+            err=True)
+    else:
+        return value
+
+
+def get_workflow_name_and_run_number(workflow_name):
+    """Return name and run_number of a workflow.
+
+    :param workflow_name: String representing Workflow name.
+
+        Name might be in format 'reana.workflow.123' with arbitrary
+        number of dot-delimited substrings, where last substring specifies
+        the run number of the workflow this workflow name refers to.
+
+        If name does not contain a valid run number, name without run number
+        is returned.
+    """
+    # Try to split a dot-separated string.
+    try:
+        name, run_number = workflow_name.rsplit('.', maxsplit=1)
```

Comment:
the `maxsplit` is not keyword argument in python 2, if you go for blah.rsplit('.', 1) it will work for both 2 and 3